### ISSUE TITLE
According to docs, local-repo is deprecated, so new-build shouldn't support it

### ISSUE THREAD

**Issue by ezyang** [haskell/cabal#3748]
The docs say:

> This can be used to access repositories on the local file system. However, the
> layout of these local repositories is different from the layout of remote
> repositories, and usage of these local repositories is deprecated.

But new-build supports it! So is it deprecated? If so let's remove it.

Feature was added in 0a2fb2356995051e1aa2c6d3ce9daa8d8affd284

CC @edsko (who added the doc entry) @dcoutts

**Comment by dcoutts** [comment]
True, though we should also check that local repos do work ok in the new secure code path.

**Comment by phadej** [comment]
If nobody takes ownership (i.e. assigns to themselves, and promises to deliver this year) before *Thursday 2019-12-05`, I'll move this issue to 3.4.

**Comment by phadej** [comment]
Related https://github.com/haskell/cabal/issues/6359

**Comment by phadej** [comment]
remilestoning for 3.4
